### PR TITLE
bpo-12414: Update code_sizeof() to take in account co_extras

### PR DESCRIPTION
**Pull request by corona10**
bpo-12414: code_sizeof() update to take in account co_extras

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://cpython-devguide.readthedocs.io/pullrequest.html#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Objects/codeobject.c [hunk 1]
```diff
@@ -452,10 +452,15 @@ static PyObject *
...
     Py_ssize_t res;
-
     res = _PyObject_SIZE(Py_TYPE(co));
+    _PyCodeObjectExtra *co_extra = (_PyCodeObjectExtra*) co->co_extra;
+
     if (co->co_cell2arg != NULL && co->co_cellvars != NULL)
         res += PyTuple_GET_SIZE(co->co_cellvars) * sizeof(Py_ssize_t);
+
+    if (co_extra != NULL)
+	res += co_extra->ce_size * sizeof(co_extra->ce_extras[0]);
```
Comment (serhiy-storchaka) [inline Objects/codeobject.c]: Is this indentation correct?

Comment (corona10) [reply Objects/codeobject.c]: @serhiy-storchaka 
Oh! I will fix this indent! 
Is there any tool for auto-formatting to solve this kind of indentation problem? 
e.g make format

Comment (serhiy-storchaka) [reply Objects/codeobject.c]: No tools for auto-formatting are used with CPython sources. You should follow PEP 7 and good sense.

**Comment by corona10** [comment]
@serhiy-storchaka Done! Thanks for the tip. PTAL

**Comment by corona10** [comment]
FYI, I sent to @betswaliszewski about the CLA issue that I didn't create my account on bugs.python.org at that time. I created it today. And this issue will be solved.

**Comment by corona10** [comment]
(CLA signed: http://bugs.python.org/user25975)

**Comment by corona10** [comment]
@serhiy-storchaka Ready for review!

File: Objects/codeobject.c [hunk 2]
```diff
@@ -452,10 +452,15 @@ static PyObject *
 code_sizeof(PyCodeObject *co, void *unused)
 {
     Py_ssize_t res;
-
     res = _PyObject_SIZE(Py_TYPE(co));
```
Comment (serhiy-storchaka) [inline Objects/codeobject.c]: If remove an empty line between declaration and assignment it is worth to merge them in one declaration with initialization.

**Comment by serhiy-storchaka** [comment]
Add an entry in `Misc/NEWS`. Include "Patch by Your Name."

And I think the test should be updated. We have no way to create `co_extra`, but if it is created automatically by the interpreter the test will fail. It should be weaked, use `>=` rather than strong `==`.

**Comment by corona10** [comment]
@serhiy-storchaka 
I update codes except for test codes.

Question:
I understood your intent about unit test.
You mean tests should go this way.
```
assertGreater(sys.getsizeof(code_obj), sizeofstruct)
```
I can not find current tests. It would be I am not familiar with the CPython codebase
Is there any sys.getsizeof() test exists on `test_code.py`?

**Comment by louisom** [comment]
@corona10 You can found it at `test_descr.py`, `test_dict.py`, `test_mmap.py`, `test_bytes.py`, for example.

**Comment by serhiy-storchaka** [comment]
`sys.getsizeof()` tests for basic types are located in `test_sys.py`.

You can just add some large value to the result in `code_sizeof()`, run all tests and see what tests are failed.

File: Misc/NEWS [hunk 3]
```diff
@@ -10,6 +10,10 @@ What's New in Python 3.7.0 alpha 1?
 Core and Builtins
 -----------------
 
+- bpo-12414: sys.getsizeof() on a code object now returns the sizes 
+  which includes the code struct and sizes of objects which it references
+  patch by Dong-hee Na
```
Comment (serhiy-storchaka) [inline Misc/NEWS]: Add periods after the sentences. Capitalize the first letter of "Patch".

**Comment by corona10** [comment]
@serhiy-storchaka Done! PTAL

File: Misc/ACKS [hunk 4]
```diff
@@ -1738,3 +1738,4 @@ evilzero
 Dhushyanth Ramasamy
 Subhendu Ghosh
 Sanjay Sundaresan
+Dong-hee Na
```
Comment (serhiy-storchaka) [inline Misc/ACKS]: Names are ordered in rough alphabetical order of surnames. Your name should be placed together with the names starting with D or N (I don't know what is your given name and what is your surname).

**Comment by corona10** [comment]
@serhiy-storchaka Thank you for your response. I updated the ACK file.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution Dong-hee!

Now you can cherry-pick b4dc6af to 3.6.

**Pull request merged**